File: src/DBuilder.Geometry/LabelPositionInfo.cs
// ABOUTME: LabelPositionInfo ported from UDB Source/Core/Geometry/LabelPositionInfo.cs.
// ABOUTME: Plain data — anchor point and clearance radius for sector label placement.

/*
 * Copyright (c) 2007 Pascal vd Heiden, www.codeimp.com
 * This program is released under GNU General Public License
 */

namespace DBuilder.Geometry;

public struct LabelPositionInfo
{
    public Vector2D position;
    public double radius;

    public LabelPositionInfo(Vector2D position, double radius)
    {
        this.position = position;
        this.radius = radius;
    }
}

File: rust/crates/dbuilder-geometry/src/label_position_info.rs
// ABOUTME: LabelPositionInfo ported from DBuilder LabelPositionInfo.cs (UDB LabelPositionInfo.cs).
// ABOUTME: Plain data; anchor point and clearance radius for sector label placement.

/*
 * Copyright (c) 2007 Pascal vd Heiden, www.codeimp.com
 * This program is released under GNU General Public License
 */

use crate::vector2d::Vector2D;

#[derive(Clone, Copy, Debug, Default, PartialEq)]
pub struct LabelPositionInfo {
    pub position: Vector2D,
    pub radius: f64,
}

impl LabelPositionInfo {
    pub fn new(position: Vector2D, radius: f64) -> LabelPositionInfo {
        LabelPositionInfo { position, radius }
    }
}
